test/scenarios/salesloft-person-custom-fields.yml

```diff
@@ -0,0 +1,12 @@
+scenario: SalesLoft Person
+description: This tests the creation and deletion of a salesloft person
+
+steps:
+- step: create or update a salesloft person
+  data:
+    person:
+        email_address: customfieldssalesloft@sample.com
+        custom_fields: 
```

Comment:
- [ ] Although this technically works in scenario yml, we should avoid using nested structures in `MAP` type fields, and swallow a little additional complexity translating the field during creation.

We could either support that by making the field name a little special (e.g. `custom_fields.NicksTest123`), or handle it in step code somehow (e.g. hard-coding standard fields and, when constructing the payload to create the Person, dynamically setting it as a custom field if the field name is not in the list of "standard" fields.)

Reply:
This is noted. For now the easiest we can do is to create a new `map` type for custom_fields than to make it `smart` to handle standard fields.

Reply:
Should we just create _**two**_ map objects or let the step accept 2 parameters?

Reply:
A scenario file example:
![image](https://user-images.githubusercontent.com/10269901/66181584-f5d5d480-e6a3-11e9-9ac0-f219c4a7ed87.png)

Reply:
My hope is to also avoid the separate `customFields` map, and that we can support a scenario like this:

```yml
- step: create or update a salesloft person
  data:
    person:
      email_address: test@example.com
      NicksTest123: this is custom
```

...Where, in the create step, we can reference a hard-coded list to determine where in the JSON payload to stick a value.

Here's the "standard" fields list, extracted from [SalesLoft docs here](https://developers.salesloft.com/api.html#!/People/post_v2_people_json)

```javascript

// Writeable fields (acceptable on Person create)
// Assume any other key is a custom field, and supply on custom_fields object.
const personStandardWriteableFields = [
  'email_address',
  'secondary_email_address',
  'personal_email_address',
  'first_name',
  'last_name',
  'phone',
  'phone_extension',
  'mobile_phone',
  'home_phone',
  'linkedin_url',
  'title',
  'city',
  'state',
  'country',
  'work_city',
  'work_state',
  'work_country',
  'person_company_name',
  'person_company_website',
  'person_company_industry',
  'job_seniority',
  'do_not_contact', // boolean
  'locale',
  'personal_website',
  'twitter_handle',
  'account_id',
  'owner_id',
  'import_id',
  'person_stage_id',
  'autotag_date', // boolean
];
```

In the case above, you could check

```typescript
const personFieldKey = 'NicksTest123';
const personCreatePayload: Record<string, any> = { custom_fields: {} };

// Write to custom_fields object if non-standard field, otherwise to root object.
if (personStandardWriteableFields.indexOf(personFieldKey) === -1) {
  personCreatePayload.custom_fields[personFieldKey] = value;
} else {
  personCreatePayload[personFieldKey] = value;
}
```

Reply:
I get you. But, should we really be hard coding like this for the fields? If this is fine with you, then I'll go ahead and change the implementation.

Reply:
I have to imagine, because they are quite tied to CRM systems like Salesforce, this is an API that they do not change often.